pkg/auth/dependency/userprofile/userprofile.go

```diff
@@ -46,47 +32,13 @@ type Store interface {
 }
 
 func (u UserProfile) MarshalJSON() ([]byte, error) {
-	return json.Marshal(u.ToMap())
+	return json.Marshal(u.Data)
```

Comment:
Why is the marshal and unmarshal function only use `u.Data` but not the whole `u`?

Reply:
Originally we had record alike`user.profile`, that is:
```
        "profile": {
            "_access": null,
            "_created_at": "2019-02-18T08:09:33.190914Z",
            "_created_by": "e44f80ee-06af-4dd9-89dd-42735ef63792",
            "_id": "user/e44f80ee-06af-4dd9-89dd-42735ef63792",
            "_ownerID": "e44f80ee-06af-4dd9-89dd-42735ef63792",
            "_recordID": "e44f80ee-06af-4dd9-89dd-42735ef63792",
            "_recordType": "user",
            "_type": "record",
            "_updated_at": "2019-02-18T08:09:33.190914Z",
            "_updated_by": "e44f80ee-06af-4dd9-89dd-42735ef63792",
            "email": "123456@example.com",
            "username": 123456
        },
```
For the latest design spec, we decide to make user `metadata` contains changeable attributes only, that is:
```
        "metadata": {
            "email": "123456@example.com",
            "username": 123456
        },
```
which is the `u.Dta